dozer/cogs/voice.py

```diff
@@ -27,6 +28,64 @@ async def on_voice_state_update(self, member, before, after):
                 if len(config) != 0:
                     await member.add_roles(member.guild.get_role(config[0].role_id))
 
+    @Cog.listener('on_voice_state_update')  # Used for auto PTT
+    async def on_PTT_check(self, member, before, after):
+        """Handles voice activity when members join/leave voice channels"""
+        # skip this if we have no perms to edit voice channel
+        total_users = 0
+        if member.guild.me.guild_permissions.manage_channels and before.channel != after.channel:
+            # determine if it's a join/leave event as well.
+            # before and after are voice states
+            if before.channel is not None:
+                # leave event, take role
+                voice_channel = before.channel
+                total_users = len(before.channel.members)
+                config = await AutoPTT.get_by(channel_id=before.channel.id)
+            if after.channel is not None:
+                # join event, give role
+                voice_channel = after.channel
+                total_users = len(after.channel.members)
+                config = await AutoPTT.get_by(channel_id=after.channel.id)
```

Comment:
If a user moves between voice channels, both `before.channel` and `after.channel` will be non-null. It looks like PTT won't be updated on `before.channel` in that case.